apps/dashboard/src/components/stack-companion/feature-request-board.tsx

```diff
@@ -0,0 +1,479 @@
+    fetchFeatureRequests().then(() => {
+      // Successfully refreshed
+    }).catch((error) => {
+      console.error('Failed to refresh feature requests:', error);
+    });
+  };
+
+  // Handle upvote
+  const handleUpvote = async (postId: string) => {
+    if (upvotingIds.has(postId)) return; // Prevent double-clicking
+
+    setUpvotingIds(prev => new Set(prev).add(postId));
+
+    const wasUpvoted = userUpvotes.has(postId);
+
+    // Optimistically update local state
+    setUserUpvotes(prev => {
+      const newSet = new Set(prev);
+      if (wasUpvoted) {
+        newSet.delete(postId);
+      } else {
+        newSet.add(postId);
+      }
+      return newSet;
+    });
+
+    // Optimistically update upvote count
+    setExistingRequests(prev => prev.map(request =>
+      request.id === postId
+        ? {
+          ...request,
+          upvotes: wasUpvoted ? Math.max(0, request.upvotes - 1) : request.upvotes + 1
+        }
+        : request
+    ));
+
+    try {
+      const authJson = await user.getAuthJson();
+      const response = await fetch(`${baseUrl}/api/v1/internal/feature-requests/${postId}/upvote`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Stack-Project-Id': 'internal',
+          'X-Stack-Access-Type': 'client',
+          'X-Stack-Access-Token': authJson.accessToken || '',
+          'X-Stack-Publishable-Client-Key': getPublicEnvVar('NEXT_PUBLIC_STACK_PUBLISHABLE_CLIENT_KEY') || '',
+        },
+        body: JSON.stringify({}),
+      });
+
+      if (response.ok) {
+        // Refresh the list to get updated upvote counts from server
+        await fetchFeatureRequests();
+      } else {
+        console.error('Failed to upvote feature request');
+        // Revert optimistic updates on failure
+        setUserUpvotes(prev => {
+          const newSet = new Set(prev);
+          if (wasUpvoted) {
+            newSet.add(postId);
+          } else {
+            newSet.delete(postId);
+          }
+          return newSet;
+        });
+        setExistingRequests(prev => prev.map(request =>
+          request.id === postId
+            ? {
+              ...request,
+              upvotes: wasUpvoted ? request.upvotes + 1 : Math.max(0, request.upvotes - 1)
+            }
+            : request
+        ));
+      }
+    } catch (error) {
+      console.error('Error upvoting feature request:', error);
+      // Revert optimistic updates on failure
+      setUserUpvotes(prev => {
+        const newSet = new Set(prev);
+        if (wasUpvoted) {
+          newSet.add(postId);
+        } else {
+          newSet.delete(postId);
+        }
+        return newSet;
+      });
+      setExistingRequests(prev => prev.map(request =>
+        request.id === postId
+          ? {
+            ...request,
+            upvotes: wasUpvoted ? request.upvotes + 1 : Math.max(0, request.upvotes - 1)
+          }
+          : request
+      ));
```

Comment:
let's keep throwing the error here so we log it & get it on sentry etc

```suggestion
      // Revert optimistic updates on failure
      setUserUpvotes(prev => {
        const newSet = new Set(prev);
        if (wasUpvoted) {
          newSet.add(postId);
        } else {
          newSet.delete(postId);
        }
        return newSet;
      });
      setExistingRequests(prev => prev.map(request =>
        request.id === postId
          ? {
            ...request,
            upvotes: wasUpvoted ? request.upvotes + 1 : Math.max(0, request.upvotes - 1)
          }
          : request
      ));
      
      throw error;
```